topo/node/alpine/alpine.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package alpine
+
+import (
+	"context"
+	"fmt"
+
+	log "k8s.io/klog/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/pointer"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apb "github.com/openconfig/kne/proto/alpine"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/kne/topo/node"
+)
+
+func New(nodeImpl *node.Impl) (node.Node, error) {
+	if nodeImpl == nil {
+		return nil, fmt.Errorf("nodeImpl cannot be nil")
+	}
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	cfg := defaults(nodeImpl.Proto)
+	nodeImpl.Proto = cfg
+	n := &Node{
+		Impl: nodeImpl,
+	}
+	return n, nil
+}
+
+type Node struct {
+	*node.Impl
+}
+
+const (
+	DefaultInitContainerImage = "us-west1-docker.pkg.dev/kne-external/kne/networkop/init-wait:ga"
+)
+func ToEnvVar(kv map[string]string) []corev1.EnvVar {
+	var envVar []corev1.EnvVar
+	for k, v := range kv {
+		envVar = append(envVar, corev1.EnvVar{
+			Name:  k,
+			Value: v,
+		})
+	}
+	return envVar
+}
+func ToResourceRequirements(kv map[string]string) corev1.ResourceRequirements {
+	r := corev1.ResourceRequirements{
+		Requests: map[corev1.ResourceName]resource.Quantity{},
+	}
+	if v, ok := kv["cpu"]; ok {
+		r.Requests["cpu"] = resource.MustParse(v)
+	}
+	if v, ok := kv["memory"]; ok {
+		r.Requests["memory"] = resource.MustParse(v)
+	}
+	return r
+}
+
+func (n *Node) Create(ctx context.Context) error {
+	if err := n.ValidateConstraints(); err != nil {
+		return fmt.Errorf("node %s failed to validate node with errors: %s", n.Name(), err)
+	}
+	if err := n.CreatePod(ctx); err != nil {
+		return fmt.Errorf("node %s failed to create pod %w", n.Name(), err)
+	}
+	if err := n.CreateService(ctx); err != nil {
+		return fmt.Errorf("node %s failed to create service %w", n.Name(), err)
+	}
+	return nil
+}
+
+// CreatePod creates a Pod for the Node based on the underlying proto.
+func (n *Node) CreatePod(ctx context.Context) error {
+	pb := n.Proto
+
+	initContainerImage := pb.Config.InitImage
+	if initContainerImage == "" {
+		initContainerImage = DefaultInitContainerImage
+	}
+	sonicName := pb.Name
+	sonicImage := pb.Config.Image
+	sonicCommand := pb.Config.Command
+	sonicArgs := pb.Config.Args
+
+	alpineContainers := []corev1.Container {
+		{
+			Name:            sonicName,
```

Comment:
nit: it looks this is the only place these variables are used, can simplify by doing `Name: pb.Name`

Reply:
Ack